### PR TITLE
Make novendor feature the default

### PR DESCRIPTION
**Pull request by mmullin-halcyon**
See: https://github.com/libbpf/libbpf-sys/pull/64

The "static" feature will now get libbpf-sys to do all the static compilation of the libraries required by libbpf rather than the application developer statically compiling these libraries themselves, or relying on the distribution to provide static versions.

The default feature will no longer link libbpf statically.  All libbpf libraries, including libbpf, will now be dynamically linked to distribution provided shared libraries

CHANGELOG NOTE #1: feature "static" allows libelf and zlib to be statically linked. Completely static binaries can now be compiled via the command `RUSTFLAGS='-C target-feature=+crt-static' cargo build --target x86_64-unknown-linux-gnu`.
CHANGELOG NOTE #2: "static" feature will not work with musl.

Thank you for considering a contribution!

In order to streamline review experience for contributors and reviewers, please
be sure to *read* and *follow* the [Contributor's Guide][cont-guide]. It
lays out basic best practices, which, if followed will reduce unnecessary back
and forth and, ultimately, minimize the time it takes to get your change into
the library.

[cont-guide]: https://github.com/libbpf/libbpf-rs/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by mmullin-halcyon** [comment]
RE: https://github.com/libbpf/libbpf-rs/pull/498

**Comment by mmullin-halcyon** [comment]
Note on Changelog #2.  I do not believe that libbpf ever compiled with musl.  musl-gcc does not "play well" with Linux Kernel headers, and libbpf uses some of the headers that musl-gcc does not play well with.  I believe that Alpine Linux has always required the "libbpf-sys/novendor" feature (now it should simply compile with default).

**Comment by mmullin-halcyon** [comment]
Posting this in anticipation of approval of the libbpf-sys change.

File: libbpf-rs/Cargo.toml [hunk 1]
```diff
@@ -15,9 +15,8 @@ keywords = ["bpf", "ebpf", "libbpf"]
 maintenance = { status = "actively-developed" }
 
 [features]
-# When turned on, link against system-installed libbpf instead of building
-# and linking against vendored libbpf sources
-novendor = ["libbpf-sys/novendor"]
+# When turned on, compile and link all libbpf library dependencies statically
+# When turned off, dynamically link all libbpf library dependencies
```
Comment (danielocfb) [inline libbpf-rs/Cargo.toml:19]: Let's please also mention that `static` implies vendoring of all C library dependencies? That is crucial from my perspective.

**Comment by danielocfb** [comment]
I think we are still blocked on waiting for a new `libbpf-sys` release before we can move forward here :-(

**Pull request closed without merging**